logdetective/server/database/models/koji.py

```diff
@@ -0,0 +1,139 @@
+from datetime import datetime, timedelta, timezone
+import enum
+from sqlalchemy import Column, BigInteger, DateTime, Enum, ForeignKey, Integer
+from sqlalchemy.orm import relationship
+from sqlalchemy.exc import OperationalError
+import backoff
+
+from logdetective.server.config import SERVER_CONFIG
+from logdetective.server.compressors import LLMResponseCompressor
+from logdetective.server.database.models.metrics import AnalyzeRequestMetrics
+from logdetective.server.database.base import Base, transaction, DB_MAX_RETRIES
+
+
+class TaskNotFoundError(Exception):
+    """Exception raised when a task is not found"""
+
+
+class TaskNotAnalyzedError(Exception):
+    """Exception raised when a task analysis is still in progress"""
+
+
+class TaskAnalysisTimeoutError(Exception):
+    """Exception raised when a task analysis has timed out"""
+
```

Comment:
wdyt about putting all exceptions into a separate module?

Reply:
I'm not opposed, but as they're somewhat generically-named, I figured keeping them in the `logdetective.server.koji` hierarchy meant that they'd be unlikely to get confused for other tasks.

If we move them, I'll probably rename them to be `KojiTask...` for clarity.